### PR TITLE
Code schemes

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Hi @as2388 take a look at the coding schemes. pending https://github.com/AfricasVoices/Project-ICRAF/pull/1

### CODE REVIEW

File: code_schemes/constituency.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
+{
+    "SchemeID": "Scheme-4c5b955d",
+    "Name": "Constituency",
+    "Version": "0.0.0.1",
+    "Codes": [
+      {
+        "CodeID": "code-bd610414",
+        "CodeType": "Normal",
+        "DisplayText": "Mwea Constituency",
```
Comment (as2388) [inline code_schemes/constituency.json]: Do you need to include "Constituency" all of the display texts?

(I'm not qualified to know the answer, just making sure there's a reason)

File: code_schemes/current_practises.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+    "SchemeID": "Scheme-8baa86c3",
+    "Name": "Current practices used to improve soil quality",
```
Comment (as2388) [inline code_schemes/current_practises.json]: Is this a follow-up question?

Wondering if we can have a more succinct scheme name, but don't have any suggestions.

File: code_schemes/ws_correct_dataset.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        "CodeType": "Normal",
+        "DisplayText": "constituency",
+        "StringValue": "constituency",
+        "NumericValue": 3,
+        "VisibleInCoda": true
+      },
+      {
+        "CodeID": "code-5c2c99ab",
+        "CodeType": "Normal",
+        "DisplayText": "livelihood",
```
Comment (as2388) [inline code_schemes/ws_correct_dataset.json:33]: Are all of the schemes above mentioned in this one?

Comment (as2388) [reply code_schemes/ws_correct_dataset.json:33]: Also the display texts here don't match the scheme names you used earlier. If these display texts are enough, then why not use these as the scheme names?

**Review by as2388 — CHANGES_REQUESTED** [review]

File: code_schemes/constituency.json (hunk 1 above)
Comment (IsaackMwenda) [reply code_schemes/constituency.json]: Actually there is no need.

File: code_schemes/ws_correct_dataset.json (hunk 3 above)
Comment (IsaackMwenda) [reply code_schemes/ws_correct_dataset.json:33]: Let me standardise these thanks!

**Comment by IsaackMwenda** [comment]
@as2388 Kindly re-review this. Thanks!

File: code_schemes/reasons_for_new _practises.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
+{
+    "SchemeID": "Scheme-405967ea",
+    "Name": "Reasons",
```
Comment (as2388) [inline code_schemes/reasons_for_new _practises.json]: Can this be something like "New Practices Reasons"?

File: code_schemes/reasons_for_new _practises.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (as2388) [inline code_schemes/reasons_for_new _practises.json]: Remove the space from the filename

File: code_schemes/new_practises.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
+{
+    "SchemeID": "Scheme-048fe72c",
+    "Name": "New practices",
```
Comment (as2388) [inline code_schemes/new_practises.json]: What is the difference between this and new practices reasons?

File: code_schemes/upper_tana_practises.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
+{
+    "SchemeID": "Scheme-9c7b500b",
+    "Name": "Upper Tana practises",
```
Comment (as2388) [inline code_schemes/upper_tana_practises.json]:
```suggestion
    "Name": "Upper Tana Practices",
```

File: code_schemes/ws_correct_dataset.json [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        "CodeID": "code-NR-5e3eee23",
+        "CodeType": "Control",
+        "ControlCode": "NR",
+        "DisplayText": "NR (not reviewed)",
+        "NumericValue": -40,
+        "StringValue": "NR",
+        "VisibleInCoda": false
+      }
+    ]
+  }
```
Comment (as2388) [inline code_schemes/ws_correct_dataset.json:136]: Shouldn't practices be in this file?

**Review by as2388 — CHANGES_REQUESTED** [review]

File: code_schemes/new_practises.json (hunk 6 above)
Comment (IsaackMwenda) [reply code_schemes/new_practises.json]: This has the labels of the new practices that the farmers are willing to implement and the reasons explain the motivation to implement the new practices.

File: code_schemes/current_practises.json [hunk 9]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (as2388) [inline code_schemes/current_practises.json]: practices with a c or a with an s? It's different in the file name/scheme name

File: code_schemes/new_practises.json (hunk 6 above)
Comment (as2388) [reply code_schemes/new_practises.json]: I see. Does the new practices scheme need to support multi-coding?

File: code_schemes/reasons_for_new_practises.json [hunk 10]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (as2388) [inline code_schemes/reasons_for_new_practises.json]: practices

File: code_schemes/upper_tana_practises.json [hunk 11]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (as2388) [inline code_schemes/upper_tana_practises.json]: practices

**Review by as2388 — APPROVED** [review]
When you've addressed my comments can you request secondary review from @lukechurch and from someone in RDA?

**Comment by IsaackMwenda** [comment]
> When you've addressed my comments can you request secondary review from @lukechurch and from someone in RDA?
@as2388  new_practices need to support multi coding. Thanks

File: code_schemes/reasons_for_new_practises.json (hunk 10 above)
Comment (IsaackMwenda) [reply code_schemes/reasons_for_new_practises.json]: Thanks updated.

**Comment by IsaackMwenda** [comment]
> When you've addressed my comments can you request secondary review from @lukechurch and from someone in RDA?

Will do that once have merged the yes-no branch https://github.com/AfricasVoices/Project-ICRAF/pull/3 into this.

**Comment by lukechurch** [comment]
@IsaackMwenda Have you addressed all of Alexander's comments? If so, please could you reply to his questions?

File: code_schemes/new_practises.json (hunk 6 above)
Comment (IsaackMwenda) [reply code_schemes/new_practises.json]: No, new practices and new practices reasons will be two different datasets

File: code_schemes/reasons_for_new _practises.json (hunk 4 above)
Comment (IsaackMwenda) [reply code_schemes/reasons_for_new _practises.json]: Updated thanks!

File: code_schemes/ws_correct_dataset.json (hunk 8 above)
Comment (IsaackMwenda) [reply code_schemes/ws_correct_dataset.json:136]: Yes have updated to include all the follow up questions thanks!

File: code_schemes/current_practises.json (hunk 9 above)
Comment (IsaackMwenda) [reply code_schemes/current_practises.json]: updated to use practices consistently.

File: code_schemes/upper_tana_practises.json (hunk 11 above)
Comment (IsaackMwenda) [reply code_schemes/upper_tana_practises.json]: updated thanks.

File: code_schemes/current_practises.json (hunk 2 above)
Comment (IsaackMwenda) [reply code_schemes/current_practises.json]: Yes, this is a follow up question.  Name updated to Current practices

File: code_schemes/reasons_for_new _practises.json (hunk 5 above)
Comment (IsaackMwenda) [reply code_schemes/reasons_for_new _practises.json]: updated thanks!

File: code_schemes/upper_tana_practises.json (hunk 7 above)
Comment (IsaackMwenda) [reply code_schemes/upper_tana_practises.json]: Updated

**Comment by IsaackMwenda** [comment]
> @IsaackMwenda Have you addressed all of Alexander's comments? If so, please could you reply to his questions?

Hi @lukechurch all updated PTAL thanks!

File: code_schemes/constituency.json [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+        "MatchValues": [
+          "ndia"
+        ]
+      },
+      {
+        "CodeID": "code-c7ec0beb",
+        "CodeType": "Normal",
+        "DisplayText": "Tetu",
+        "StringValue": "tetu",
+        "NumericValue": 3,
```
Comment (lukechurch) [inline code_schemes/constituency.json]: These shouldn't have the same numeric value

File: code_schemes/livelihood.json [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+        "VisibleInCoda": true,
+        "MatchValues" : [
+          "business"
+        ]
+      },
+      {
+        "CodeID": "code-7e1dd056",
+        "CodeType": "Normal",
+        "DisplayText": "nurse",
+        "NumericValue": 2,
```
Comment (lukechurch) [inline code_schemes/livelihood.json]: NumericValue repeat

File: configurations/pipeline_config.json [hunk 14]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (lukechurch) [inline configurations/pipeline_config.json]: This should be in a different PR really - it's not adjusting the code schemes as the PR suggests it should.

**Review by lukechurch — CHANGES_REQUESTED** [review]

File: code_schemes/constituency.json (hunk 12 above)
Comment (IsaackMwenda) [reply code_schemes/constituency.json]: Thanks @lukechurch have updated!

File: code_schemes/livelihood.json (hunk 13 above)
Comment (IsaackMwenda) [reply code_schemes/livelihood.json]: Thanks have updated.

File: configurations/pipeline_config.json (hunk 14 above)
Comment (IsaackMwenda) [reply configurations/pipeline_config.json]: Thank you this was on a previous pending review.

**Comment by lukechurch** [comment]
Thanks @IsaackMwenda - Please can you file an issue against the core infrastructure that we should be validating when we load the code schemes that all the numeric and string values are unique?

**Review by lukechurch — APPROVED** [review]
